docs/deployment/cli.md

```diff
@@ -23,664 +23,254 @@ specific language governing permissions and limitations
-                           e/deployment/config.html
-     -e,--executor <arg>   DEPRECATED: Please use the -t option instead which is
-                           also available with the "Application Mode".
-                           The name of the executor to be used for executing the
-                           given job, which is equivalent to the
-                           "execution.target" config option. The currently
-                           available executors are: "remote", "local",
-                           "kubernetes-session", "yarn-per-job", "yarn-session".
-     -t,--target <arg>     The deployment target for the given application,
-                           which is equivalent to the "execution.target" config
-                           option. The currently available targets are:
-                           "remote", "local", "kubernetes-session",
-                           "yarn-per-job", "yarn-session", "yarn-application"
-                           and "kubernetes-application".
-
-  Options for yarn-cluster mode:
-     -m,--jobmanager <arg>            Set to yarn-cluster to use YARN execution
-                                      mode.
-     -yid,--yarnapplicationId <arg>   Attach to running YARN session
-     -z,--zookeeperNamespace <arg>    Namespace to create the Zookeeper
-                                      sub-paths for high availability mode
-
-  Options for default mode:
-     -m,--jobmanager <arg>           Address of the JobManager to which to
-                                     connect. Use this flag to connect to a
-                                     different JobManager than the one specified
-                                     in the configuration. Attention: This
-                                     option is respected only if the
-                                     high-availability configuration is NONE.
-     -z,--zookeeperNamespace <arg>   Namespace to create the Zookeeper sub-paths
-                                     for high availability mode
-{% endhighlight %}
+## Advanced CLI
+ 
+### REST API
+
+The Flink cluster can be also managed using the [REST API]({% link ops/rest_api.md %}). The commands 
+described in previous sections are a subset of what is offered by Flink's REST endpoints. Therefore, 
+tools like `curl` can be used to get even more out of Flink.
+
+### Selecting Deployment Targets
+
+Flink supports different deployment backends like [Kubernetes]({% link deployment/resource-providers/native_kubernetes.md %}) 
+or [YARN]({% link deployment/resource-providers/yarn.md %}) which are described in more detail in the 
+Resource Provider section. You can specify the mode you're Flink cluster is running in through the 
+`--target` parameter. It supports the following values:
+* `local`
+* `remote`
+* `kubernetes-session`
+* `kubernetes-application`
+* `yarn-per-job`
+* `yarn-session`
+* `yarn-application`
```

Comment:
I think it would be good to document that `*-application` can only be used with `run-application` and the others only with `run`.

What is actually the difference between `flink list --target yarn-per-job` and `flink list --target yarn-session`? I guess one need to specify somehow that one wants to talk to a Yarn cluster. But the whether it is a session, a per-job or a application cluster should not really matter. @kl0u can you maybe chime in here?

Reply:
There is no difference between the two commands that you mentioned @tillrohrmann . Both tell the CLI that it will be "talking" to a YARN cluster so it will go essentially through the `YarnClusterDescriptor`. 

As for documenting that each of the `run` and `run-application` are compatible with a subset with the available `execution.targets`, I agree.

Reply:
@kl0u Could you come up with a proposal for the `Selecting Deployment Targets` section? 😇 You might have more insights it might want to add a bit more? Or is the action-to-exection-target mapping the only important information that should be added to this section?

Reply:
Yes, it would be great to add some more information to this section.